### ISSUE TITLE
outlier_detection not working

### ISSUE THREAD

**Issue by Guang91** [spacetelescope/jwst#4337]
I'm running the pipeline on some MIRI images. The outlier_detection (stage 3) successfully detects some CR outliers. But it only changes the DQ values of CR pixels to 4. Then in the step (resampling), these pixels are considered "good" and the mosaic image still have these CR outliers. That is to say, outlier_detection does not affect the final science image. 

I think it's better to flag the CR pixels as a higher value (say 14) in the DQ map. Then the resampling step can reject the CR outliers. 

Is it a reasonable approach?    

Well... After setting it to 14, I realized too many pixels are identified as CR. The outlier_detection step seems to be *very* ineffective.

**Comment by jdavies-st** [comment]
Hi @Guang91, I think this is probably fixed in #4726.  Could you verify that it is working properly for your simulated MIRI data?  We were not setting DO_NOT_USE properly, and this should now be fixed.

Generally, any pixels that are not to be used in a pipeline step are set to 1, i.e. DO_NOT_USE.  Every other bit flag is informational only.  It looks like we don't have this documented very well in the docs.  But to find out what is what:

```
In [21]: from jwst.datamodels.dqflags import pixel

In [22]: pixel
Out[22]: 
{'GOOD': 0,
 'DO_NOT_USE': 1,
 'SATURATED': 2,
 'JUMP_DET': 4,
 'DROPOUT': 8,
 'RESERVED_1': 16,
 'RESERVED_2': 32,
 'RESERVED_3': 64,
 'RESERVED_4': 128,
 'UNRELIABLE_ERROR': 256,
 'NON_SCIENCE': 512,
 'DEAD': 1024,
 'HOT': 2048,
 'WARM': 4096,
 'LOW_QE': 8192,
 'RC': 16384,
 'TELEGRAPH': 32768,
 'NONLINEAR': 65536,
 'BAD_REF_PIXEL': 131072,
 'NO_FLAT_FIELD': 262144,
 'NO_GAIN_VALUE': 524288,
 'NO_LIN_CORR': 1048576,
 'NO_SAT_CHECK': 2097152,
 'UNRELIABLE_BIAS': 4194304,
 'UNRELIABLE_DARK': 8388608,
 'UNRELIABLE_SLOPE': 16777216,
 'UNRELIABLE_FLAT': 33554432,
 'OPEN': 67108864,
 'ADJ_OPEN': 134217728,
 'UNRELIABLE_RESET': 268435456,
 'MSA_FAILED_OPEN': 536870912,
 'OTHER_BAD_PIXEL': 1073741824,
 'REFERENCE_PIXEL': 2147483648}
```

There's a separate set of flags only used in the stage 1 pipeline with the 4D data cubes used to ID groups.

```
In [23]: from jwst.datamodels.dqflags import group

In [24]: group
Out[24]: {'GOOD': 0, 'DO_NOT_USE': 1, 'SATURATED': 2, 'JUMP_DET': 4, 'DROPOUT': 8}
```

Anyway, let us know if there are any remaining issues.

**Comment by Guang91** [comment]
Now, it appears working properly. Thanks a lot, @jdavies-st
